woocommerce/woocommerce-ios
# [Order Details] Show Collect Payment button to all merchants

Part of: #7003 


### Description


This removes the checks for `cardPresentPaymentsEligibility` from Order Details, and instead shows the Collect Payment button for any order where the `datePaid` is nil, regardless of where the merchant's store is based or what the status is of their payment gateway accounts.

The Collect Payment button opens the Payment Method screen, as added in #7110.

This PR fixes the issue where the Collect Payment button would remain on screen after taking a payment.

Outstanding issue: some order statuses probably shouldn't allow payment, e.g. cancelled, failed... but they currently do.

### Testing instructions


With a store in a non-CPP country (i.e. not US or Canada):

1. On the Order tab, add an order and go through to take payment
2. Observe that the collect payment button shows, and tapping it takes the user to choose a payment method from the options Cash or Payment Link
3. Use the cash method, and observe that the order is marked as paid.

With a store in a CPP country, but no WCPay/Stripe extension installed:
1. Repeat the above steps, but choose the Card payment option
2. Observe that you're shown the onboarding screen advising you to install WCPay

With a store in a CPP country, with WCPay installed and set up:
1. Repeat the above steps
2. Observe that you're able to connect to a reader and take a card payment.


### Screenshots



https://user-images.githubusercontent.com/2472348/174825975-62a0f0c1-040b-43f8-a807-3cdd666c3d8d.mp4



---
